packages/stream-caip10-link-handler/src/caip10-link-handler.ts

```diff
@@ -153,7 +157,9 @@ export class Caip10LinkHandler implements StreamHandler<Caip10Link> {
     state: StreamState
   ): Promise<StreamState> {
     // TODO: Assert that the 'prev' of the commit being applied is the end of the log in 'state'
-    const proof = (await context.ipfs.dag.get(commit.proof, { timeout: IPFS_GET_TIMEOUT })).value
+    const proof = StreamUtils.isExpandedCommit(commit)
+      ? commit.proof
+      : (await context.ipfs.dag.get(commit.proof, { timeout: IPFS_GET_TIMEOUT })).value
```

Comment:
We should go through the dispatcher rather than accessing IPFS directly

Reply:
yeah, I think it would be fine to put the Dispatcher into the Context.  Ideally we'd get rid of the IPFS handle out of Context at the same time, but don't worry about that if it's used in too many places